src/differentiation/compute_jacobian_ad.jl

```diff
@@ -80,8 +88,8 @@ function forwarddiff_color_jacobian(f,x::AbstractArray{<:Number},jac_cache::Forw
 
     vecx = vec(x)
     
-    ncols=length(x)
-    J = jac_prototype isa Nothing ? false .* x .* x' : zero(jac_prototype)
+    J = jac_prototype isa Nothing ? (sparsity isa Nothing ? false .* dx .* x' : zeros(eltype(x),size(sparsity))) : zero(jac_prototype)
```

Comment:
`zeros` gives an `Array`

Reply:
I think when jac_prototype is not specified, J should be an Array.

Reply:
true, makes sense.